docs/contracts/transient-storage.rst

```diff
@@ -0,0 +1,61 @@
+.. index:: ! transient storage
+
+.. _transient-storage:
+
+*****************
+Transient Storage
+*****************
+
+Transient storage is another data location besides memory, storage, calldata
+(and return-data and code) which was introduced alongside its respective opcodes
+``TSTORE`` and ``TLOAD`` by `EIP-1153 <https://eips.ethereum.org/EIPS/eip-1153>`_.
+This new data location behaves as a key-value store similar to storage with the main
+difference being that data in transient storage is not permanent, but is scoped to
+the current transaction only, after which it will be reset to zero. The values on
+transient storage are never deserialized from storage or serialized to storage,
+thus the gas cost of operating in transient storage is much cheaper,
+since it doesn't require disk access.
+
+Currently, the compiler can parse ``transient`` as a data location, however it is not
+defined as a keyword of the language yet. This means that the use of ``transient``
+is backwards-compatible and does not break previous code that eventually used it as an identifier.
```

Comment:
This is just a small technical detail. I'd put it in a note later in the text - it should be less prominent.